src/script/lua_api/l_object.cpp

```diff
@@ -678,6 +678,81 @@ int ObjectRef::l_get_properties(lua_State *L)
 	return 1;
 }
 
+// set_observers(self, observers)
+int ObjectRef::l_set_observers(lua_State *L)
+{
+	GET_ENV_PTR;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		throw LuaError("Invalid ObjectRef");
+
+	// Reset object to "unmanaged" (sent to everyone)?
+	if (lua_isnoneornil(L, 2)) {
+		ServerActiveObject *parent = sao->getParent();
+		if (!parent->m_observer_names.has_value())
```

Comment:
missing null check?

Reply:
`parent->m_observer_names` is a `std::optional` here, which should be default-initialized to be empty. The `.has_value()` here should always be valid.

Reply:
surely `parent` is null when there is no parent

Reply:
Ah, you're referring to the `parent`. Sorry, yeah that's right.